### DISCUSSION TITLE
FaceSwap

### DISCUSSION THREAD

**Discussion by LilithDragoness** [lllyasviel/stable-diffusion-webui-forge#1818] [Q&A]
How can I use Faceswap in Forge in a way that’s similar to Fooocus'? Specifically, I’m looking for settings comparable to these: Type: FaceSwap, Stop At: 1, Weight: 0.75.

**Comment by DarkNeo88** [comment]
just install the extension https://github.com/Ethereum-John/sd-webui-forge-faceswaplab

**Reply by juujaab** [reply]
this extension is amazing!i used very well

**Reply by stripealipe** [reply]
I mean, this doesn't work when installed?! Just like ReActor etc, it doesn't appear in the tab section...

**Reply by gushf8** [reply]
Could you please zip the folder for that extension and send it to me via Google Drive?

**Reply by Wontfallo** [reply]
both labs and react not found at github any more?

**Comment by 00BlackBeard00** [comment]
Why i got this error when try to build a face?
No such file or directory: 'C:\\StabilityMatrix-win-x64\\Data\\Packages\\Stable Diffusion WebUI Forge\\extensions\\sd-webui-faceswaplab\\references\\woman.png'

**Reply by juujaab** [reply]
yeah,i have same problem

**Reply by nitinmukesh** [reply]
@00BlackBeard00 @juujaab 

See my comment in this topic at the bottom.

**Reply by WillyamBradberry** [reply]
FaceSwapLab - ERROR - Failed to build checkpoint [Errno 2] No such file or directory: ..\\sd-forge\\webui\\extensions\\sd-webui-faceswaplab\\references\\man.png'

**Reply by WillyamBradberry** [reply]
> FaceSwapLab - ERROR - Failed to build checkpoint [Errno 2] No such file or directory: ..\sd-forge\webui\extensions\sd-webui-faceswaplab\references\man.png'

But if I do the same in A1111 it works and the model appears in the list in Forge after. Very inconvinient way, because I was about to delete A1111 to save more space on my laptop.

**Reply by TigermanUK** [reply]
I still found this error in the forge version of faceswap for people looking to fix it.
This is the quick dirty FIX. locate your ..\sd-forge\extensions folder inside you will see your sd-webui-forge-faceswaplab folder.
In ..\sd-forge\extensions make a new folder named "sd-webui-faceswaplab".  Next copy the folder references that is inside sd-webui-forge-faceswaplab and paste it into your newly created sd-webui-faceswaplab folder.
Double check go into sd-webui-faceswaplab\references there should be two files man.png and woman.png.  These where the missing files that have the wrong path in the forge version of faceswap you have just corrected.

**Comment by wisdombox44** [comment]
İt is already in forgewebui tabs just install and use it natively. Spaces amazing no need extention
![360066561-cf0c7095-c52f-4617-971a-e5e740feb297](https://github.com/user-attachments/assets/9a22583e-dd5a-44e0-9415-febb65ca4f32)

**Comment by kalle07** [comment]
reactor is more useful for me

https://github.com/Gourieff/sd-webui-reactor

**Reply by stripealipe** [reply]
Doesn't work in Forge either though?!

**Reply by tazztone** [reply]
it works. i pretty much tested all functions in forge. 
![image](https://github.com/user-attachments/assets/1bef6370-956e-4240-916f-5cabd18210b4)

**Reply by stripealipe** [reply]
> it works. i pretty much tested all functions in forge. ![image](https://private-user-images.githubusercontent.com/62671577/369412810-1bef6370-956e-4240-916f-5cabd18210b4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcxMDgxMDksIm5iZiI6MTcyNzEwNzgwOSwicGF0aCI6Ii82MjY3MTU3Ny8zNjk0MTI4MTAtMWJlZjYzNzAtOTU2ZS00MjQwLTkxNmYtNWNhYmQxODIxMGI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIzVDE2MTAwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMwZDZmY2VlYTNiYjBlOTE1MWE4YjMzYjY0ZjMyNzFiMjEyYTFlNjY0MzZkZDM3Y2ExMmY2ZjY4Njg2OWZjNjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ZyNyRCC7VBEOK2WLrSHzVJm8S2UumxKIgezU8808LHo)

Yeah, i'm getting the following error and i've tried everything lol... might have to just reinstall everything from scratch maybe?!

**ModuleNotFoundError: No module named 'scripts.faceswaplab_utils'**

If anyone has ever had this and has a solution, shout!! Thanks...

**** UPDATE
ok, seems my install was borked... I had Python installed on a system level from way back and I think that was maybe messing with file paths etc. Anyways, removed that and reinstalled Forge and all is good now.

**Reply by TigermanUK** [reply]
If you have reactor and faceswaplab(fsl) installed at the same time. you can swap upto 4 faces at once. Three in faceswaplab and the fourth in reactor.  So there is a use case for FSL's even if you find ReActor more useful.

**Comment by nitinmukesh** [comment]
I installed the extension
https://github.com/Ethereum-John/sd-webui-forge-faceswaplab

Getting this error


```
2024-09-20 00:06:31,827 - FaceSwapLab - INFO - Choosen gender : Gender.AUTO
2024-09-20 00:06:31,833 - FaceSwapLab - INFO - Build Face with 1 images
2024-09-20 00:06:31,868 - FaceSwapLab - INFO - Load analysis model det_size=(640, 640), det_thresh=0.5, providers = ['CPUExecutionProvider'], will take some time. (> 30s)
100%|██████████████████████████████████████████████████████████████████████| 281857/281857 [00:22<00:00, 12810.01KB/s]
Loading (640, 640) analysis model (first time is slow): 100%|████████████████████████| 1/1 [00:25<00:00, 25.80s/model]
2024-09-20 00:06:57,695 - FaceSwapLab - INFO - ('download_path: '
 'C:\\usable\\webui_forge_cu121_torch231\\webui\\models\\faceswaplab\\analysers\\models\\buffalo_l\n'
 'Downloading '
 'C:\\usable\\webui_forge_cu121_torch231\\webui\\models\\faceswaplab\\analysers\\models\\buffalo_l.zip '
 'from '
 'https://github.com/deepinsight/insightface/releases/download/v0.7/buffalo_l.zip...\n'
 "Applied providers: ['CPUExecutionProvider'], with options: "
 "{'CPUExecutionProvider': {}}\n"
 'find model: '
 'C:\\usable\\webui_forge_cu121_torch231\\webui\\models\\faceswaplab\\analysers\\models\\buffalo_l\\1k3d68.onnx '
 "landmark_3d_68 ['None', 3, 192, 192] 0.0 1.0\n"
 "Applied providers: ['CPUExecutionProvider'], with options: "
 "{'CPUExecutionProvider': {}}\n"
 'find model: '
 'C:\\usable\\webui_forge_cu121_torch231\\webui\\models\\faceswaplab\\analysers\\models\\buffalo_l\\2d106det.onnx '
 "landmark_2d_106 ['None', 3, 192, 192] 0.0 1.0\n"
 "Applied providers: ['CPUExecutionProvider'], with options: "
 "{'CPUExecutionProvider': {}}\n"
 'find model: '
 'C:\\usable\\webui_forge_cu121_torch231\\webui\\models\\faceswaplab\\analysers\\models\\buffalo_l\\det_10g.onnx '
 "detection [1, 3, '?', '?'] 127.5 128.0\n"
 "Applied providers: ['CPUExecutionProvider'], with options: "
 "{'CPUExecutionProvider': {}}\n"
 'find model: '
 'C:\\usable\\webui_forge_cu121_torch231\\webui\\models\\faceswaplab\\analysers\\models\\buffalo_l\\genderage.onnx '
 "genderage ['None', 3, 96, 96] 0.0 1.0\n"
 "Applied providers: ['CPUExecutionProvider'], with options: "
 "{'CPUExecutionProvider': {}}\n"
 'find model: '
 'C:\\usable\\webui_forge_cu121_torch231\\webui\\models\\faceswaplab\\analysers\\models\\buffalo_l\\w600k_r50.onnx '
 "recognition ['None', 3, 112, 112] 127.5 127.5\n"
 'set det-size: (640, 640)\n')
C:\usable\webui_forge_cu121_torch231\webui\venv\lib\site-packages\insightface\utils\transform.py:68: FutureWarning: `rcond` parameter will change to the default of machine precision times ``max(M, N)`` where M and N are the input matrix dimensions.
To use the future default and silence this warning we advise to pass `rcond=None`, to keep using the old, explicitly pass `rcond=-1`.
  P = np.linalg.lstsq(X_homo, Y)[0].T # Affine matrix. 3 x 4
2024-09-20 00:06:57,938 - FaceSwapLab - INFO - Int Gender : 0
2024-09-20 00:06:57,939 - FaceSwapLab - ERROR - Failed to build checkpoint [Errno 2] No such file or directory: 'C:\\usable\\webui_forge_cu121_torch231\\webui\\extensions\\sd-webui-faceswaplab\\references\\woman.png'
Traceback (most recent call last):
  File "C:\usable\webui_forge_cu121_torch231\webui\extensions\sd-webui-forge-faceswaplab\scripts\faceswaplab_swapping\face_checkpoints.py", line 76, in build_face_checkpoint_and_save
    reference_preview_img = Image.open(
  File "C:\usable\webui_forge_cu121_torch231\webui\venv\lib\site-packages\PIL\Image.py", line 3236, in open
    fp = builtins.open(filename, "rb")
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\usable\\webui_forge_cu121_torch231\\webui\\extensions\\sd-webui-faceswaplab\\references\\woman.png'

```

**Reply by nitinmukesh** [reply]
SOLUTION

Install using git clone method.
Use Notepad++ to search all occurrences of sd-webui-faceswaplab and replace with sd-webui-forge-faceswaplab in *.py files.
Under Directory specify the full path of the sd-webui-forge-faceswaplab folder

![image](https://github.com/user-attachments/assets/f56ff26b-c8a6-48e9-916b-47cd769425b9)

**Reply by juujaab** [reply]
i only replace the *py in checkpoint, it came to work,thx

**Reply by WillyamBradberry** [reply]
Yeah, that works. As well as if I just rename folders.

**Reply by seventenjake** [reply]
Just did this and still getting the error. Any ideas?

FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\linde\\ForgeFlux\\webui_forge_cu121_torch231\\webui\\extensions\\sd-webui-faceswaplab\\references\\woman.png'
2025-01-11 16:05:35,105 - FaceSwapLab - INFO - Choosen gender : Gender.FEMALE
2025-01-11 16:05:35,106 - FaceSwapLab - INFO - Build Face with 4 images
2025-01-11 16:05:35,602 - FaceSwapLab - INFO - Int Gender : 0
2025-01-11 16:05:35,603 - FaceSwapLab - ERROR - Failed to build checkpoint [Errno 2] No such file or directory: 'C:\\Users\\linde\\ForgeFlux\\webui_forge_cu121_torch231\\webui\\extensions\\sd-webui-faceswaplab\\references\\woman.png'
Traceback (most recent call last):
  File "C:\Users\linde\ForgeFlux\webui_forge_cu121_torch231\webui\extensions\sd-webui-forge-faceswaplab\scripts\faceswaplab_swapping\face_checkpoints.py", line 76, in build_face_checkpoint_and_save
    reference_preview_img = Image.open(
  File "C:\Users\linde\ForgeFlux\webui_forge_cu121_torch231\system\python\lib\site-packages\PIL\Image.py", line 3236, in open
    fp = builtins.open(filename, "rb")

**Comment by MigCurto** [comment]
I'm experimenting ReActor and it works ok. Dont have experience with this but the results seem good.

**Reply by kalle07** [reply]
it works best in bit distance, and if you use adetailer first with "old" or "mature" face its a bit better

if u have a lora from your "favorit" u can use adetailer alone (copy the lora insice adetailer prompt), push up denoise 0.6-0.7

**Reply by kalle07** [reply]
it was not the same error ,but for me worket another face-swap model

![grafik](https://github.com/user-attachments/assets/406b3561-ef3b-4ef8-9df0-8fe884e2a491)

Unfortunately i downt know where you can save this setting, the save button is only for CUDA

**Comment by gushf8** [comment]
I think so, but censored versions.

**Reply by SvenErik1968** [reply]
Go to the folder `extensions\sd-webui-reactor-sfw\scripts\`
open the file `reactor_sfw.py` in a text editor, change 
`SCORE = 0.965`
to
`SCORE = 1.000`
and it is "uncensored"...
